🎯 Add Delegate Banner to Tokens Table

fairlighteth opened the pull request:
## Overview
This PR introduces a new delegate banner component to encourage users to delegate their (v)COW tokens. The banner is strategically placed in the tokens table and provides a direct link.

## Key Changes
- ✨ Added new `Delegate` component with a visually appealing two-tone design
- 🎨 Created custom delegate-cow SVG asset for visual appeal
- 📱 Implemented responsive layout with proper grid system
- 🔗 Added UTM tracking for delegate link analytics
- 🧹 Improved overall layout and responsiveness of the tokens table

## Technical Details
- Follows TypeScript best practices and project architecture guidelines
- Utilizes existing UI components and styling patterns
- Implements proper responsive behavior using styled-components
- Maintains consistency with the existing design system

## Testing
Please verify:
- [ ] Banner displays correctly in the tokens table
- [ ] Responsive behavior works on different screen sizes
- [ ] Delegate link includes proper UTM parameters
- [ ] Visual consistency with existing UI components

## Screenshots
<img width="449" alt="Screenshot 2025-02-11 at 16 41 29" src="https://github.com/user-attachments/assets/cacc5a90-415c-4d36-a83c-4f16a7708191" />
<img width="1564" alt="Screenshot 2025-02-11 at 16 41 18" src="https://github.com/user-attachments/assets/43efd037-ac93-4926-af16-76fa33f0753b" />
<img width="1562" alt="Screenshot 2025-02-11 at 16 41 12" src="https://github.com/user-attachments/assets/ce88b397-6480-412e-96aa-def48b7e6554" />
<img width="784" alt="Screenshot 2025-02-11 at 16 41 02" src="https://github.com/user-attachments/assets/469b0852-464f-4f4e-af51-063d6adcf832" />
<img width="1440" alt="Screenshot 2025-02-11 at 16 40 49" src="https://github.com/user-attachments/assets/b20393ba-e8be-4d80-89cc-c0018446d87f" />
<img width="1518" alt="Screenshot 2025-02-11 at 16 40 43" src="https://github.com/user-attachments/assets/092a37e5-e545-4b01-8b4d-143a6d215770" />



<!-- This is an auto-generated comment: release notes by coderabbit.ai -->
## Summary by CodeRabbit

- **New Features**
  - Introduced a new delegation feature with an interactive banner that encourages users to delegate tokens.
  - Enabled additional content rendering within the tokens display for enhanced flexibility.
  - Added a new `Delegate` component to facilitate token delegation.

- **Style**
  - Updated layout spacing in the side menu and tokens display for a cleaner, more balanced interface.
  - Refined the presentation of governance and account page elements with refreshed typography and a modern grid layout.
  - Enhanced the use of CSS variables for styling consistency across components.
  - Added a new `CloseButton` component with hover effects for improved user experience.

- **Bug Fixes**
  - Improved logic for determining the swap button's disabled state, ensuring correct functionality.
  
- **Chores**
  - Introduced constants for banner identifiers and storage keys to enhance maintainability and organization.
<!-- end of auto-generated comment: release notes by coderabbit.ai -->

alfetopito reviewed (approved):
Looks good.
Just pending Elena's comments.

fairlighteth commented:
Did some refactoring:

## Changes
- Moved UI storage keys to component-specific locations (`ClosableBanner/constants.ts`)
- Fixed duplicate state management in `TokensTable` (removed local atom, using shared `closableBannersStateAtom`)
- Improved code organization following component-centric approach
- Fixed import ordering and removed unnecessary dependencies

## Architecture
- Better DDD alignment (component-specific constants with components)
- Proper dependency direction (apps → libs)
- Enhanced state sharing across components
- Improved type safety and maintainability

## Testing
Verify banner state persistence and dismissal functionality works across the app

elena-zh commented:
Hey @fairlighteth , I don't see any changes on the UI using the preview links. Could you please check whether your new changes were deployed there?

fairlighteth commented:
## Some updates

## Changes
- Enhanced delegate banner with flexible mobile layout support via new `rowOnMobile` prop
- Updated delegate documentation link to new comprehensive article
- Improved table styling with proper z-index, background, and mobile responsiveness
- Fixed border color consistency using UI variables
- Enhanced mobile table display and layout handling
- Removed redundant styling code

## Some updates:
- Added TypeScript prop `rowOnMobile` for layout control
- Updated delegate URL to `cow.fi/learn/cow-dao-launches-delegate-program-for-token-holders`
- Maintained consistent UI variable usage throughout styling
- Improved component reusability

## Testing Notes
- Verify delegate banner display on desktop/mobile
- Check table styling across screen sizes
- Confirm delegate documentation link and UTM parameters
- Test table header visibility while scrolling

[Screenshots pending]

alfetopito commented on apps/cowswap-frontend/src/common/constants/banners.ts:
```diff
@@ -0,0 +1,7 @@
+// Banner IDs for various dismissable banners across the app
+export const BANNER_IDS = {
+  DELEGATE: 'delegate-banner-v1',
```
Can you do it camelCased like the others?

Ok, there's a mix of camelCase with/without underscore and one snake_case...
But let's at least not introduce a kebab-case to the mix 🍢

elena-zh commented:
Hey @fairlighteth , thank you!
Mobile view looks better now!
However, still not sure about the banner view in the table: 
![image](https://github.com/user-attachments/assets/b66921a1-2613-4212-9005-c13abdba8302)

WDYT about placing this dismissible card above the Tokens table? 
![image](https://github.com/user-attachments/assets/0339c2eb-33f1-41fc-bab1-f32d4226f33b)

And still I'd opt for expanding the delegation card when there is an odd number of cards on the page
![image](https://github.com/user-attachments/assets/91d97c45-5e03-4e6c-8245-e9aa7d3ac341)

Thanks!

elena-zh reviewed (approved):
I still don't like how the Tokens page looks like in a mobile view, but given the fact that we are short of time/resources, approving it.
